featurizers/fingerprints.py

```diff
@@ -261,3 +270,99 @@ def _calculate_fingerprint(
         }
 
         return np.array([GetErGFingerprint(x, **fp_args) for x in X])
+
+
+class E3FP(FingerprintTransformer):
+    def __init__(
+        self,
+        confgen_params: dict,
+        fprint_params: dict,
+        is_folded: bool = False,
+        fold_bits: int = 1024,
+        standardise: bool = False,
+        random_state: int = 0,
+        n_jobs: int = 1,
+        verbose: int = 0,
+    ):
+        if "first" not in confgen_params:
+            confgen_params["first"] = fprint_params.get("first", 1)
+        if "first" not in fprint_params:
+            fprint_params["first"] = confgen_params.get("first", 1)
+
+        # Get necessary default values if not specified
+        from e3fp.conformer.generate import (
+            NUM_CONF_DEF,
+            POOL_MULTIPLIER_DEF,
+            RMSD_CUTOFF_DEF,
+            MAX_ENERGY_DIFF_DEF,
+            FORCEFIELD_DEF,
+            SEED_DEF,
+        )
+
+        confgen_params["num_conf"] = confgen_params.get(
+            "num_conf", NUM_CONF_DEF
+        )
+        confgen_params["pool_multiplier"] = confgen_params.get(
+            "pool_multiplier", POOL_MULTIPLIER_DEF
+        )
+        confgen_params["rmsd_cutoff"] = confgen_params.get(
+            "rmsd_cutoff", RMSD_CUTOFF_DEF
+        )
+        confgen_params["max_energy_diff"] = confgen_params.get(
+            "max_energy_diff", MAX_ENERGY_DIFF_DEF
+        )
+        confgen_params["forcefield"] = confgen_params.get(
+            "forcefield", FORCEFIELD_DEF
+        )
```

Comment:
In general, I think that we should use separate parameters. I know that E3FP library is ugly and uses dicts, but we should do better.

Reply:
I mean separate parameters should be passed, and assigned to attributes. This is also necessary, since Scikit-learn relies on that behavior. See https://scikit-learn.org/stable/developers/develop.html